dali/base/dadfs.cpp

```diff
@@ -13425,10 +13448,20 @@ IPropertyTreeIterator *deserializeFileAttrIterator(MemoryBuffer& mb, unsigned nu
             else
                 sizeDiskSize = file->getPropInt64(getDFUQResultFieldName(DFUQRForigsize), 0);
             double sizeGB = sizeDiskSize / ((double)1024 * 1024 * 1024);
-            __int64 numDiskWrites = file->getPropInt64(getDFUQResultFieldName(DFUQRFnumDiskReads), 0);
-            __int64 numDiskReads = file->getPropInt64(getDFUQResultFieldName(DFUQRFnumDiskWrites), 0);
-            double atRestCost, accessCost;
-            calcFileCost(nodeGroup, sizeGB, fileAgeDays, numDiskWrites, numDiskReads, atRestCost, accessCost);
+            double accessCost = 0.0;
+            if (hasReadWriteCostFields(*file))
+            {
+                // Newer files have readCost and writeCost attributes
+                accessCost = cost_type2money(file->getPropInt64(getDFUQResultFieldName(DFUQRFreadCost)) + file->getPropInt64(getDFUQResultFieldName(DFUQRFwriteCost)));
+            }
+            else // Calc access cost from numDiskRead & numDiskWrites for Legacy files
+            {
+                cost_type legacyReadCost = getLegacyReadCost(*file, nodeGroup);
+                __int64 numDiskWrites = file->getPropInt64(getDFUQResultFieldName(DFUQRFnumDiskWrites), 0);
+
+                accessCost = legacyReadCost + calcFileAccessCost(nodeGroup, numDiskWrites, 0);
+            }
+            double atRestCost = calcFileAtRestCost(nodeGroup, sizeGB, fileAgeDays);
             file->setPropReal(getDFUQResultFieldName(DFUQRFcost), atRestCost+accessCost);
             file->setPropReal(getDFUQResultFieldName(DFUQRFatRestCost), atRestCost);
             file->setPropReal(getDFUQResultFieldName(DFUQRFaccessCost), accessCost);
```

Comment:
hm, baffled, I wrote a comment on last review submission, but can't find it now..

As the code stands (in this PR), "readCost" and "writeCost" are published as top-level file attributes in cost_type format
e.g.:
```
        cost_type curReadCost = money2cost_type(calcFileAccessCost(file, 0, curDiskReads));
        file->addAttrValue(getDFUQResultFieldName(DFUQRFreadCost), legacyReadCost + curReadCost);
```

But then here, in attributes seen by clients using getDFAttributesTreeIterator, it adds "cost", "atRestCost" and "accessCost" which as reals (converted via cost_type2money).
The net result is that all these attributes will be visible to a caller of getDFAttributesTreeIterator, but some will be in one format and some will be in another.

We should make sure they are consistently in the same format.
I'm not sure it matters which format as long as consistent (I suspect ideally they'd remain as cost_type, until a client usage case needs them).
Either change the new "readCost" "writeCost" attributes so they're published in "money" format, and change calculations accordingly.
Or, keep them as cost_type, but then existing code which expects them as 'money' (e.g. getCost) and the esp service code, would need to change.